### ISSUE TITLE
Finite intersections

### ISSUE THREAD

**Issue by benjub** [metamath/set.mm#2743]
While editing the comment of http://us2.metamath.org/mpeuni/neiptopreu.html, I noticed that http://us2.metamath.org/mpeuni/df-fi.html defines nonempty finite intersections, whereas its comment, and the section title, say "Finite intersections".  (Side note: I think this is sloppy and wouldn't give a good image of our work if some outsider landed on that page and saw that.)  The quick fix is to modify the comment and section title, but I would suggest instead to include the empty intersection.

Of course, the abstract empty intersection is the universal class, which is a proper class so cannot be collected in a class.  But in the context of df-fi, we typically deal with intersections of families of subsets, so it is natural to make the definition
```
df-fi $a |- fi = ( x e. _V |-> { z | E. y e. ( ~P x i^i Fin ) z = ( |^| y i^i U. x ) } ) $.
```
This would entail modifying some proofs using it, so it would require some work, but it is probably worth it (the case of the empty intersection would no longer have to be treated separately).

Incidentally, this is what I had done in http://us2.metamath.org/mpeuni/bj-mooreset.html and http://us2.metamath.org/mpeuni/df-bj-moore.html (for arbitrary intersections) before noticing the issue of df-fi.

By the way: the theorem bj-mooreset, which is right before the definition df-bj-moore, serves as a kind of justification for it: it says that all classes sastisfying some predicate are actually sets, so can be collected in a class; in other words, the definition of the class does not "miss" any class satisfying the given predicate.  I don't know if there are other occurrences of this in set.mm.

**Edti**: I will do the minimal change of adding "nonempty" to the comment of df-fi, while we look at the possible implications of a definition change.  I haven't noticed problematic implications for now but the range of uses is vast...

**Comment by digama0** [comment]
I disagree that the comment is imprecise. The definition clearly has no special case for the empty intersection, and it is indeed the collection of all finite intersections of the input set. The empty intersection doesn't exist so it is not included in that quantifier.

If you want to make a definition for the class of finite intersections intersected with some set, you are welcome to but that would be a different definition.

**Comment by benjub** [comment]
> it is indeed the collection of all finite intersections of the input set.

This is at best misleading. set.mm talks of the empty intersection (e.g., ~int0) even if it is a proper class so is a "non-exsitent object" to use terms from free logic.

I found this problem while reading the comment of http://us2.metamath.org/mpeuni/neiptopreu.html (I just rephrased it in the last PR), so my concern is a practical one. When we talk of a family of subsets stable by intersections (or finite intersections), like a topology, Moore collection or sigma-algebra, we include the empty intersection (or else we specify "nonempty (finite) intersections") which is the whole set. My proposal is in line with the current use in set.mm (though uncommon outside of set.mm) of considering a topology `J` and then defining its base set as `U. J`.

So again, the concern is practical: having easier applications with fewer particular cases.  If it happens that set.mm has more applications using nonempty finite applications, then indeed let's stay with the current definition (but I would say: at least, mention it in the comment).
